cdap-app-fabric/src/main/java/io/cdap/cdap/scheduler/ScheduleNotificationSubscriberService.java

```diff
@@ -206,7 +208,26 @@ protected void processNotification(ProgramScheduleStoreDataset scheduleStore,
         LOG.warn("Ignore notification that doesn't have a schedule {} associated with, {}", scheduleId, notification);
         return;
       }
-      jobQueue.addNotification(record, notification);
+      try {
+        jobQueue.addNotification(record, notification);
+      } catch (Exception e) {
+        emitScheduleJobNotificationFailureMetrics();
+        throw e;
+      }
+    }
+
+    private void emitScheduleJobNotificationFailureMetrics() {
+      if (metricsCollectionService == null) {
+        return;
+      }
+      MetricsContext collector = metricsCollectionService.getContext(getContext());
+      collector.increment("schedulejob.notification.failure", 1);
```

Comment:
Then we should also make this schedulejob.failure right? At least, all failures with the same metric name can be aggregated.

Reply:
this metric signifies the failure to process(push the notification to JobQueue) a message(job schedule notification) from TMS. If the processing fails, the message is not removed from TMS and the message processing is retired. So, there can be multiple failure metrics("schedulejob.notification.failure") corresponding to failure of scheduling one job("schedulejob.failure").